plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -150,6 +151,23 @@ internal fun Package.toSpdxPackage(
         SpdxPackageVerificationCode(packageVerificationCodeValue = it)
     }
 
+    val resolvedLicenseExpressions = licenseInfoResolver.resolveLicenseInfo(id).filterExcluded()
+        .applyChoices(ortResult.getPackageLicenseChoices(id))
+        .applyChoices(ortResult.getRepositoryLicenseChoices())
+
+    val declaredPackageLicenses = resolvedLicenseExpressions.filter(LicenseView.ONLY_DECLARED)
+
+    val patterns = LicenseFilePatterns.getInstance()
+    val detectedPackageLicenses = resolvedLicenseExpressions.filterTo(mutableSetOf()) { licenseInfo ->
+        licenseInfo.locations.any {
+            FileMatcher.match(patterns.allLicenseFilenames, it.location.path, ignoreCase = true)
```

Comment:
TODO: This should also work work packages in sub-directories which contain their own license.